File: DotNetGrc/Grc/IR/Node/Stmt/StmtBlock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Grc.IR.Quads;

namespace Grc.Ast.Node.Stmt
{
	public partial class StmtBlock : StmtBase
	{
		public override IEnumerable<Quad> IR
		{
			get
			{
				IEnumerable<Quad> ie = null;

				foreach (StmtBase s in stmts)
					ie = ie == null ? s.IR : ie.Concat(s.IR);

				return ie == null ? this.ir : ie.Concat(this.ir);
			}
		}
	}
}

File: Parser/src/main/java/k31/grc/ast/node/stmt/StmtBlock.java
package k31.grc.ast.node.stmt;

import java.util.LinkedList;
import java.util.List;

import k31.grc.ast.visitor.Visitor;

public class StmtBlock extends StmtBase {

	private List<StmtBase> stmts;

	public List<StmtBase> getStmts() {
		return stmts;
	}

	public StmtBlock(String text) {
		super(text);

		this.stmts = new LinkedList<StmtBase>();
	}

	@Override
	public void accept(Visitor v) {

		v.visit(this);
	}
}
